### ISSUE TITLE
Small error in the README?

### ISSUE THREAD

**Issue by fennecdjay** [vrtbl/passerine#47]
Shaw pointed me to this language.
And when reading the README, I got puzzled by this:
```c
linear = m b x -> b + m * x
linear 2 3 5
-- evaluates to 13
```
the following explaination makes sense tho, but it use `3 2 5`, not `3 2 5`.

Maybe I'm mistaken.
If I'm not, I could do the PR to fix this.

**Comment by ShawSumma** [comment]
This is correct behavior.
`m b x -> b + m * x` is equivalent to `m -> b -> x -> b + m * x`
Currying is used when calling functions, left to right.

If it helps, the substitution is as follows:
1. start with the expression`linear 2 3 5`
2. load variable linear, giving `(m b x -> b + m * x) 2 3 5`
3. replace `m` with 2, giving `(b x -> b + 2 * x) 3 5`
4. replace `b` with 3, giving `(x -> 3 + 2 * x) 5`
5. replace `x` with 5, giving `3 + 2 * 5`
6. this evaluates to `13`

I can see how this is confusing tho. I have always seen it as `m * x + b` and read it as "y equals m x plus b".
It may be helpful to change it to ```
linear = m x b -> m * x + b
linear 2 5 3
-- evaluates to 13
```

**Comment by fennecdjay** [comment]
At least the explanation should match the example imo.

**Comment by slightknack** [comment]
Sorry, I was on vacation and didn't see this message until now. I think that changing `b + m * x` to `m * x + b` makes sense. I think my original intent was showing that variables could be used in any order: of course, this is pretty obvious, so it makes sense to change it. I'll reread that section and change it soon. Thanks!

**Comment by slightknack** [comment]
After doing some more thinking, I realize why I ordered that equation the way I did:

1. Because of currying, it makes sense to order arguments by amount of expected variation; what I mean by this, is if you have a higher order function like `map`, you can define the function`double` as `map { x -> x * 2 }`. Likewise, if you wanted to make a linear equation, you could do something like `my_line = linear -3 5`; that function now accepts any `x`. 

2. The reason I used the form `a + b * c` as opposed to `b * c + a` was to demonstrate operator precedence. When you think about it, the following line is just a bunch of operators:
  ```
  linear = m b x -> b + m * x
  -- same as:
  linear = (m b x -> (b + (m * x)))
  ```
  I wanted to show operator precedence was respected.
  
That's about it. There are few ways we can proceed: use the `m * x + b` form and adjust the explanation, use the current form, but make it clearer as to why it's formulated the way it is.

Another thing to note is that when linear equations are written with the addition first, the variables `a` and `b` are usually used, as in: `a + b * x`. Just something to think about :)

**Comment by slightknack** [comment]
Also, this example by far has resulted in the most questions. Perhaps it's because it's the first example, but it might be worthwhile to look for a better leading example.

**Comment by slightknack** [comment]
@ShawSumma and I have discussed this in our latest weekly meeting, and we've decided that switching this out to a `distance` function would make the most sense. We'll try to get this updated and merged in soon.

**Comment by slightknack** [comment]
Here's the distance function:
```
square = x -> x * x

distance = (x1, y1) (x2, y2) -> {
    sqrt (square (x1 - x2) + square (y1 - y2))
}

length = distance (0, 0)
```

The reason we're defining `square` instead of using `**` is because it's a simple definition that's really hard to get confused over. We build on that base by then showing how arguments are passed and functions are called, and then we show the power of function call syntax by defining `length` in terms of `distance`.

**Comment by slightknack** [comment]
We will refactor the README as a part of `0.10.0`, which will include this change.

**Comment by slightknack** [comment]
Fixed in `6f5a0cd`! Let me know if there is anything in the new version that needs to be fixed. I'm closing this for now :)
